### ISSUE TITLE
Auth issue with GHES

### ISSUE THREAD

**Issue by homer3018** [Andrew-Chen-Wang/github-wiki-action#74]
Hello,

I'm trying to use this action on my company GitHub server. From the documentation it looks like this should be fairly straight forward, and my yaml file looks like this:
```
name: Publish wiki
on:
  push:
    branches:
      - main
      - develop
    paths:
      - wiki/**
      - .github/workflows/publish-wiki.yml
concurrency:
  group: publish-wiki
  cancel-in-progress: true
permissions:
  contents: write
jobs:
  publish-wiki:
    runs-on: company-ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      # nosemgrep: yaml.github-actions.security.third-party-action-not-pinned-to-commit-sha.third-party-action-not-pinned-to-commit-sha
      - uses: Andrew-Chen-Wang/github-wiki-action@v4
        with:
          github-server-url: https://github.company.com
```

However it fails with the following error : `fatal: could not read Username for 'https://github.company.com': No such device or address`.

More specifically, here is the full log : 
```
Run Andrew-Chen-Wang/github-wiki-action@v4
  with:
    github-server-url: https://github.company.com
    strategy: clone
    repository: org/repo
    token: ***
    path: wiki
    commit-message: Update wiki 66d0df532ff1a5799302dc547247f5dd8ea37b6b
    dry-run: false
Run "${GITHUB_ACTION_PATH%/}/src/$INPUT_STRATEGY.sh"
  "${GITHUB_ACTION_PATH%/}/src/$INPUT_STRATEGY.sh"
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    INPUT_STRATEGY: clone
    INPUT_REPOSITORY: org/repo
    INPUT_GITHUB_SERVER_URL: https://github.company.com
    INPUT_TOKEN: ***
    INPUT_PATH: wiki
    INPUT_COMMIT_MESSAGE: Update wiki 66d0df532ff1a5799302dc547247f5dd8ea37b6b
    INPUT_IGNORE: 
    INPUT_DRY_RUN: false
Cloning into bare repository '/tmp/tmp.7P3kZTZ8eH'...
fatal: could not read Username for 'https://github.company.com': No such device or address
Error: Process completed with exit code 128.
```

I just changed my **company**, **org** and **repo** for the sake of it.

I suspect this is failing at the `gh auth setup-git`command, but i'm not sure.
Any hint to solve this would be appreciated.

Thanks ahead of time for your support !

**Comment by homer3018** [comment]
Should I open a pull request with that change ?

**Comment by homer3018** [comment]
Hi @jcbhmr, would you be able to help me out ?

**Comment by Andrew-Chen-Wang** [comment]
@homer3018 Have you created a sample WIki page first? Like go to the tab Wiki itself and create a Home.md first

**Comment by homer3018** [comment]
I did.

This is now solved, and here are a couple of things worth noting :
- Since V4 on this repo, I noticed changes with regards to the GitHub server URL which makes my earlier comments about the `https://` piece not relevant anymore.
- In my case since I use GitHub Enterprise, I also need to use `GH_ENTERPRISE_TOKEN`, and as a result the only thing I modified on my fork is to remove `GH_TOKEN` (see [here](https://github.com/homer3018/github-wiki-action/commit/322140081f01a7791363d18ba75b4256ac249930)). 
- Lastly and related to the above, in my yaml file I simply added an env clause like so
```
        env:
          GH_ENTERPRISE_TOKEN=${{ secrets.WIKI_ACTION_TOKEN }}
```
